integration/login_logout_test.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"fmt"
+	"os/exec"
+
+	"github.com/go-check/check"
+)
+
+func init() {
+	check.Suite(&LoginLogoutSuite{})
+}
+
+type LoginLogoutSuite struct {
+	regV2WithAuth *testRegistryV2
+}
+
+func (s *LoginLogoutSuite) SetUpSuite(c *check.C) {
+	_, err := exec.LookPath(skopeoBinary)
+	c.Assert(err, check.IsNil)
+}
+
+func (s *LoginLogoutSuite) TearDownSuite(c *check.C) {
+
+}
+
+func (s *LoginLogoutSuite) SetUpTest(c *check.C) {
+	s.regV2WithAuth = setupRegistryV2At(c, privateRegistryURL1, true, false)
+}
+
+func (s *LoginLogoutSuite) TearDownTest(c *check.C) {
+	if s.regV2WithAuth != nil {
+		//cmd := exec.Command("docker", "logout", s.regV2WithAuth)
+		//c.Assert(cmd.Run(), check.IsNil)
+		s.regV2WithAuth.Close()
+	}
+}
+
+func (s *LoginLogoutSuite) TestLoginLogout(c *check.C) {
+	wanted := "^Login Succeeded!\n$"
+	assertSkopeoSucceeds(c, wanted, "login", "--tls-verify=false", "--username="+s.regV2WithAuth.username, "--password="+s.regV2WithAuth.password, s.regV2WithAuth.url)
+	// test --get-login returns username
+	wanted = fmt.Sprintf("^%s\n$", s.regV2WithAuth.username)
+	assertSkopeoSucceeds(c, wanted, "login", "--tls-verify=false", "--get-login", s.regV2WithAuth.url)
+	// test logout
+	wanted = fmt.Sprintf("^Removed login credentials for %s\n$", s.regV2WithAuth.url)
+	assertSkopeoSucceeds(c, wanted, "logout", s.regV2WithAuth.url)
+}
+
+func (s *LoginLogoutSuite) TestCopyWithLocalAuth(c *check.C) {
+	wanted := "^Login Succeeded!\n$"
+	assertSkopeoSucceeds(c, wanted, "login", "--tls-verify=false", "--username="+s.regV2WithAuth.username, "--password="+s.regV2WithAuth.password, s.regV2WithAuth.url)
+	// copy to private registry using local authentication
+	imageName := fmt.Sprintf("docker://%s/busybox:mine", s.regV2WithAuth.url)
+	assertSkopeoSucceeds(c, "", "copy", "--dest-tls-verify=false", "docker://docker.io/library/busybox:latest", imageName)
+	// inspec from private registry
+	assertSkopeoSucceeds(c, "", "inspect", "--tls-verify=false", imageName)
+	// logout from the registry
+	wanted = fmt.Sprintf("^Removed login credentials for %s\n$", s.regV2WithAuth.url)
+	assertSkopeoSucceeds(c, wanted, "logout", s.regV2WithAuth.url)
```

Comment:
Non-blocking: This could have one more test here to verify that `inspect` now fails, i.e. that `logout` did work.